backend/app/models/analytics_model.go

```
@@ -7,49 +7,52 @@ import (
+	// Only auto-migrate URLMetric, removing the old tables
 	db.GetDB().AutoMigrate(&URLAnalytics{})
-	db.GetDB().AutoMigrate(&URLReferrer{})
-	db.GetDB().AutoMigrate(&URLEngagement{})
-	db.GetDB().AutoMigrate(&URLCountryAnalytics{})
+	db.GetDB().AutoMigrate(&URLAnalyticsHourly{})
+	db.GetDB().AutoMigrate(&URLAnalyticsDaily{})
+	db.GetDB().AutoMigrate(&URLAnalyticsMonthly{})
+
+	// Enable TimescaleDB for analytics if available
+	if db.IsTimescaleEnabled {
+		db.InitializeTimescale()
```

golangci-lint (1.64.8)

Error return value of `db.InitializeTimescale` is not checked

(errcheck)